.github/workflows/pytest-core-nompi.yml

```diff
@@ -36,102 +35,104 @@ jobs:
-              openmp: 1
-
-            - name: py38-gcc9-openmp
-              python-version: 3.8
-              gcc-version: 9
-              openmp: 1
+        - name: py36-gcc49-openmp
+          python-version: 3.6
+          os: ubuntu-16.04
+          arch: "gcc-4.9"
+          openmp: 1
+
+        - name: py37-gcc5-openmp
+          python-version: 3.7
+          os: ubuntu-16.04
+          arch: "gcc-5"
+          openmp: 1
+
+        - name: py38-gcc6-openmp
+          python-version: 3.8
+          os: ubuntu-16.04
+          arch: "gcc-6"
+          openmp: 1
+
+        - name: py36-gcc7-openmp
+          python-version: 3.6
+          os: ubuntu-16.04
+          arch: "gcc-7"
+          openmp: 1
+
+        - name: py37-gcc7-noopenmp
+          python-version: 3.7
+          os: ubuntu-16.04
+          arch: "gcc-7"
+          openmp: 0
+
+        - name: py37-gcc8-openmp
+          python-version: 3.7
+          os: ubuntu-16.04
+          arch: "gcc-8"
+          openmp: 1
+
+        - name: py38-gcc9-openmp
+          python-version: 3.8
+          os: ubuntu-16.04
+          arch: "gcc-9"
+          openmp: 1
+
+        - name: py37-osx
+          python-version: 3.7
+          os: macos-latest
+          arch: "osx"
+          openmp: 0
+
+        - name: docker-devito
+          python-version: 3.6
+          os: ubuntu-16.04
+          arch: "gcc"
+          openmp: 1
 
     steps:
     - name: Checkout devito
       uses: actions/checkout@v1
 
     - name: Set up Python ${{ matrix.python-version }}
+      if: matrix.name != 'docker-devito'
       uses: actions/setup-python@v1
       with:
         python-version: ${{ matrix.python-version }}
 
-    - name: Install GCC ${{ matrix.gcc-version }}
-      run : |
-        sudo apt-get install -y g++-${{ matrix.gcc-version }}
-
-    - name: Install dependencies
-      run: |
-        pip install --upgrade pip
-        pip install -e .
-
-    - name: Test with pytest
+    - name: Build docker image
+      if: matrix.name == 'docker-devito'
       run: |
-        pytest --cov --cov-config=.coveragerc --cov-report=xml tests/
-
-    - name: Upload coverage to Codecov
-      uses: codecov/codecov-action@v1.0.5
-      with:
-        token: ${{ secrets.CODECOV_TOKEN }}
-        name: ${{ matrix.name }}
+          docker build . --file docker/Dockerfile --tag devito_img
+          echo "::set-env name=RUN_CMD::docker run --rm --name testrun devito_img"
 
-  # OSX
-  ci-osx:
-    name: py37-clang-osx
-    runs-on: macos-latest
-
-    env:
-      DEVITO_ARCH: "osx"
-      DEVITO_OPENMP: 0
-      DEVITO_BACKEND: "core"
-
-    steps:
-    - name: Checkout devito
-      uses: actions/checkout@v1
+    - name: Install GCC ${{ matrix.arch }}
+      if: runner.os == 'linux'
+      run : |
+        sudo apt-get install -y ${{ matrix.arch }}
 
-    - name: Set up Python 3.7
-      uses: actions/setup-python@v1
-      with:
-        python-version: 3.7
+    - name: Reduce number of tests for OSX
```

Comment:
Is it worth doing this for some of the other builds? Or would it be too much?

Reply:
Could be done to speed up yes but can't trim too much or not testing enough

Reply:
OK